Lib/ensurepip/__init__.py

```diff
@@ -106,12 +103,9 @@ def _bootstrap(*, root=None, upgrade=False, user=False,
         additional_paths = []
         for project, version, py_tag in _PROJECTS:
             wheel_name = "{}-{}-{}-none-any.whl".format(project, version, py_tag)
-            whl = resources.read_binary(
-                _bundled,
-                wheel_name,
-            )
+            whl = resources.files("ensurepip") / "_bundled" / wheel_name
             with open(os.path.join(tmpdir, wheel_name), "wb") as fp:
-                fp.write(whl)
+                fp.write(whl.read_bytes())
```

Comment:
`read_bytes` is not ensured by `importlib.abc.Traversable`, it only works here because the actual implementation happens to have it, but that is not guaranteed.

It should be replaced with
```python
with whl.open('rb'):
    whl.read()
```

Reply:
Since this PR got stale the module was modified somewhat (https://github.com/python/cpython/pull/24210) 
I had to rebase and force push, but this is changed now.

Reply:
Oops, I made a mistake in the snippet!

We are opening a file, so it would be
```python
with whl.open('rb') as f:
    f.read()
```

:facepalm:

Reply:
D'oh!   Just noticed the same after a load of tests failed.

Reply:
Ugh. Well, I must have been sleepy or something, `Traversable` does define a fallback `read_bytes`, so it should be provided by the protocol. Sorry.